Issue #1747 - Tangent Improvements

latenitefilms opened the pull request:
- Added `plugins.finalcutpro.tangent.common` to make adding Tangent UI
elements easier
- Moved the Final Cut Pro Tangent modes to
`plugins.finalcutpro.tangent.modes`
- Rewrote `plugins.finalcutpro.tangent.video` to make use of new common
functions
- Tweaked `InspectorProperty` documentation
- Added Tangent Knob for Timeline Clip Height
- Added Tangent Knob for Timeline Clip Waveform Height
- Added Tangent Buttons for Timeline Clip Waveform Height
- Added Tangent Buttons for Timeline Display Options
- Added Tangent controls for Crop group
- Added Tangent controls for Distort group
- Added Tangent controls for Opacity
- Added Tangent controls for Stabilization
- Closes #1747

latenitefilms committed:
#1747

- Added Tangent Knob for Timeline Clip Height
- Added Tangent Knob for Timeline Clip Waveform Height
- Added Tangent Buttons for Timeline Clip Waveform Height
- Added Tangent Buttons for Timeline Display Options
- Added Tangent controls for Crop group
- Added Tangent controls for Distort group
- Added Tangent controls for Opacity
- Added Tangent controls for Stabilization
- Closes #1747

latenitefilms committed:
#1747

- Added `plugins.finalcutpro.tangent.common` to make adding Tangent UI
elements easier
- Moved the Final Cut Pro Tangent modes to
`plugins.finalcutpro.tangent.modes`
- Rewrote `plugins.finalcutpro.tangent.video` to make use of new common
functions
- Tweaked `InspectorProperty` documentation

randomeizer committed:
Added the `PropertyRow:extend(extendFn)` method

Allows simple extensions of the instance via an inline function.

randomeizer committed:
Added `extend(extendFn)` to the `section` property row cp.prop.

randomeizer committed:
Fixed `section:extend(...)` and added `simple:extend(...)`

randomeizer committed:
Fixed typo with `simple:extend`

randomeizer committed:
Fixed bug with `section/simple:extend(...)`

randomeizer committed:
Renamed local variables for clarity.

latenitefilms committed:
#1747

- Work in progress
- Still slowly hooking up all the Video & Audio Inspector properties to
the Tangent

randomeizer committed:
#1747
* Added a 'pixel buffer' for determining if axuielements are in a particular PropertyRow.

randomeizer committed:
#1747
* Fixed sticker issues.

latenitefilms committed:
#1747

- Added various Video & Audio Tangent controls

latenitefilms committed:
#1747

- Started building `AudioConfiguration`

randomeizer commented on src/extensions/cp/apple/finalcutpro/inspector/audio/AudioConfiguration.lua:
```diff
@@ -0,0 +1,137 @@
+-- THE MODULE:
+--
+--------------------------------------------------------------------------------
+local AudioConfiguration = Element:subclass("AudioConfiguration")
```
Just put the full path to the class in the name. And we use `ScrollArea` here because this is an `AXScrollArea`. That way we inherit all the existing functionality for ScrollArea automatically.

```suggestion
local AudioConfiguration = ScrollArea:subclass("cp.apple.finalcutpro.inspector.audio.AudioConfiguration")
```

randomeizer commented on src/extensions/cp/apple/finalcutpro/inspector/audio/AudioConfiguration.lua:
```diff
@@ -0,0 +1,137 @@
+--------------------------------------------------------------------------------
+local AudioConfiguration = Element:subclass("AudioConfiguration")
+
+function AudioConfiguration.__tostring()
```
This is unnecessary when setting the name to the full package, as suggested above. Just delete the `__tostring()` function altogether.

randomeizer commented on src/extensions/cp/apple/finalcutpro/inspector/audio/AudioConfiguration.lua:
```diff
@@ -0,0 +1,137 @@
+local log                               = require("hs.logger").new("audioConfiguration")
+
+local axutils                           = require("cp.ui.axutils")
+local Element                           = require("cp.ui.Element")
```
Since this seems to be representing an AXScrollArea, lets just extend that directly.
```suggestion
local ScrollArea                           = require("cp.ui.ScrollArea")
```

randomeizer committed:
Update src/extensions/cp/apple/finalcutpro/inspector/audio/AudioConfiguration.lua

Co-Authored-By: latenitefilms <latenitefilms@users.noreply.github.com>

latenitefilms committed:
#1747

- Resolves @randomeizer’s suggestions

latenitefilms committed:
#1747

- Fixed @stickler-ci errors

latenitefilms committed:
#1747

- Added `cp.apple.finalcutpro.inspector.audio.AudioComponent`
- Added `cp.ui.axutils.childrenInLine`

latenitefilms committed:
#1747

- Added `cp.ui.Image` and `cp.ui.TextArea`
- Changes to `AudioConfiguration` and `AudioComponent`

latenitefilms committed:
#1747

- Tweaked `cp.ui.axutils.childIndex`
- Added `cp.ui.ColorWell`
- Added various Audio & Text Tangent elements

latenitefilms committed:
#1747

- Fixed @stickler-ci errors

latenitefilms committed:
#1747

- Fixed a bug that would cause TextField’s to fail if the system’s
decimal number separator was set to “,” instead of “.”.
- Closes #1624
- Closes #1513

latenitefilms committed:
#1747

- Added `cp.ui.axutils.childrenInNextLine`
- Fixed potential bug in `cp.ui.Element:doShow()`
- Additions to `TextInspector` API
- Made `DELAY` a constant in the Tangent Common Tools
- Added more Tangent Text controls

latenitefilms committed:
#1747

- Fixed @stickler-ci bugs

latenitefilms committed:
Merge branch 'develop' into issue/1747-tangent-timeline-height

latenitefilms committed:
#1747

- Added some more Text Tangent Controls
- Tried to solve the show/hide issue in the inspector but failed
miserably

latenitefilms committed:
#1747

- Fixed @sticker-ci errors

latenitefilms committed:
Merge branch 'develop' into issue/1747-tangent-timeline-height

# Conflicts:
#	src/extensions/cp/ui/Image.lua

latenitefilms committed:
#1747

- Added `cp.dev.highlightFrame`
- Added some of the `cp.ui` modules from Timeline Index branch
- Added more Tangent controls

latenitefilms committed:
#1747

- Fixed @sticker-ci errors

latenitefilms committed:
#1747

- Added support for `NSDeleteCharacter` in
`cp.apple.commandeditor.characterStringToKeyCode`
- Changed `cp.apple.finalcutpro.main.LibrariesBrowser.filterToggle` to
`cp.apple.finalcutpro.main.LibrariesBrowser.clipFiltering`
- Fixed Rx bugs in `cp.ui.MenuButton`
- Added action to toggle the HUD Search Panel
- Added various Tangent controls
- Fixed documentation bugs in `cp.ui.ElementCache`

latenitefilms committed:
#1747

- Fixed @stickler-ci errors

latenitefilms committed:
#1747

- Fixed @stickler-ci errors

latenitefilms committed:
#1747

- Fixed @stickler-ci error
